### PR TITLE
Personal group

### PR DESCRIPTION
**Pull request by mixmix**
This PR adds the functionality needed to create + find custom links to particular groups.

It also refactors some of the existing code into `spec/group` folder ... that shoulda been a seperate PR mebee

### CODE REVIEW

File: method/index.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 
 module.exports = function Method (ssb, keystore, state) {
   return {
     group: {
       init: patient(Init(ssb, keystore, state)),
       addMember: patient(AddMember(ssb, keystore, state))
+    },
+    link: {
+      create: patient(CreateLink(ssb)),
+      findGroupByFeedId: FindByGroupByFeedId(ssb)
```
Comment (mixmix) [inline method/index.js:14]: does this need to be patient? I don't think so as the indexes query will make it patient for you

File: README.md [hunk 2]
```diff
@@ -110,6 +110,41 @@ where:
...
+
+Creates a message of type `link/feed-group` which links your feedId to a valid group. (i.e. you can only create links between your feedId and profiles at the moment)
+
+Arguments:
+- `group` *GroupId* - the id for a group you're wanting to create link to
+- `name` *String* (optional) - this adds your nickname for the group
+- `cb` *Function* - callback with signature `(err, link)` where `link` is the link message
+
+Note:
+- this link will be encrypted to the group you're linking to (i.e. link will have `recps: [groupId`])
```
Comment (chereseeriepa) [inline README.md:123]: fix the end of this line

File: README.md [hunk 3]
```diff
@@ -110,6 +110,41 @@ where:
 - `cb` *Function* - has signature `cb(err, success)`
 
 
+### `ssb.tribes.link.create({ group, name }, cb)`
+
+Creates a message of type `link/feed-group` which links your feedId to a valid group. (i.e. you can only create links between your feedId and profiles at the moment)
+
+Arguments:
+- `group` *GroupId* - the id for a group you're wanting to create link to
```
Comment (chereseeriepa) [inline README.md:118]: this sentence doesnt make sense

Comment (chereseeriepa) [reply README.md:118]: missing `a` i think?

**Pull request merged**